app/controllers/api/admin/articles_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Api::Admin::ArticlesController < ApplicationController
+  before_action :authenticate_user!
+  before_action { check_user_role() }
+
+  def index
+    if current_user.role == 'editor'
+      article = Article.where(published: false)
+      render json: article, each_serializer: Admin::Article::IndexSerializer
+    else
+      render_unauthorized
+    end
+  end
+end
+
+private
+
+def check_user_role()
+  if current_user.role == 'user' || current_user.role == 'subscriber'
```

Comment:
out of scope, just check that the user is an editor

Reply:
![Screen Shot 2020-05-28 at 16 32 16](https://user-images.githubusercontent.com/61978848/83154862-eb6bbd80-a100-11ea-9c95-9139219138ab.png)

You literally asked for this

Reply:
No, not really. I was discussing with you how the controller could potentially be going to look like with all roles in the future. 

At this point what I want you to do is.

```
class Api::Admin::ArticlesController < ApplicationController
  before_action :authenticate_user!
  before_action :admin?

  def index
      article = Article.where(published: false)
      render json: article, each_serializer: Admin::Article::IndexSerializer
  end

private

def render_unauthorized
  render json: { message: 'You are not authorized', errors: ['You are not authorized'] }, status: 401
end

def admin?
  unless current_user.role == 'editor'
    render_unauthorized
  end
end
```

Reply:
I actually don't think this is any more elegant or neat than the original solution. It still wont help with the fact that user and subscribers are always unauthorized and journalists only sometimes. 

Did a small trick, and it's still not a before action, but will allow you to skip all the checks and just say what you want to limit your privilege at. It is basically the same as in the start and at this point I must insist that it is a good solution and will easily work with additional roles and controllers. In the very last commit now I also fixed the enum to allow the addition of the subscriber without any issue.